kernel/timeout.c

```diff
@@ -83,8 +83,18 @@ static s32_t next_timeout(void)
 	return ret;
 }
 
-void z_add_timeout(struct _timeout *to, _timeout_func_t fn, s32_t ticks)
+void z_add_timeout(struct _timeout *to, _timeout_func_t fn, k_timeout_t t)
 {
+#ifdef CONFIG_LEGACY_TIMEOUT_API
+	k_ticks_t ticks = t;
+#else
+	k_ticks_t ticks = t.ticks + 1;
+
+	if (IS_ENABLED(CONFIG_TIMEOUT_64BIT) && Z_TICK_ABS(ticks) >= 0) {
+		ticks = Z_TICK_ABS(ticks) - curr_tick;
```

Comment:
seems to me that it should be `- (curr_tick + elapsed())`

Reply:
Arrgh, right as I was about to push.  Yes, this is a bug.  Will fix and update.